spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/procedures/RegisterTableProcedure.java

```diff
@@ -86,6 +87,12 @@ public InternalRow[] call(InternalRow args) {
         "Cannot handle an empty argument metadata_file");
 
     Catalog icebergCatalog = ((HasIcebergCatalog) tableCatalog()).icebergCatalog();
+    if (tableName.hasNamespace() && icebergCatalog instanceof SupportsNamespaces) {
```

Comment:
I don't think this check should be part of the procedure. Instead, such a check should be added in `registerTable()` in `BaseMetastoreCatalog` / `RestSessionCatalog`. Tests can then be added to `CatalogTests` for `RestSessionCatalog`.  Tests for registerTable in `BaseMetastoreCatalog` are spread across different catalog implementations, so tests should be added there as well